keps/sig-release/2572-release-cadence/README.md

```diff
@@ -0,0 +1,321 @@
+contributors in doing more.
+
+#### SIG Release members
+
+By applying a cadence of 3 releases per year, SIG Release members will gain a
+reduced management overhead. There are also only 3 patch releases to maintain,
+which right now can overlap up to 4. SIG Release will gain more time to ensure
+a seamless transition from the previous release team to the next one. It is
+also possible to include more shadows if the role leads conclude that this is
+appropriate.
+
+### Risks and Mitigations
+
+#### Concentrating risk
+
+In theory a reduced release cadence will cause more changes for every release.
+This means that there will be an increased risk, which would usually be split
+up into 4 dedicated milestones rather than 3.
+
+SIG Release cannot mitigate this risk directly, but is able to track and
+influence it during each release cycle. It's the responsibility of SIG Release,
+together with SIG Testing and SIG Architecture, to identify new gaps and issues
+in the release cadence and mitigate them on a case-by-case basis.
+
+#### Attention to tests
+
+This KEP does not propose any change to the release cycle itself and assumes
+that the same periods for Code and Test Freeze. Assuming that, there is an
+increased risk for flakes and test failures. It will be the responsibility of
+SIG Release to mitigate this, together with the CI signal role. If we speak
+about an overall release cycle enhancement of 3-4 weeks, then we believe that
+SIG Release is able to mitigate this risk over multiple releases.
+
+#### Attention to dependencies
+
+Having fewer releases will introduce the risk of missing dependencies—for
+example, Golang upgrades. This has to be mitigated on a case-by-case basis, in
+the same way as it is being done right now.
+
+## Design Details
+
+### Schedule Policy
+
+The exact details of the release schedule are created by the team leads before
+the actual cycle begins. SIG Release only provides a lightweight policy, which
+is defined as:
+
+1. The first Kubernetes release of a year should start at the second or third
+   week of January to provide people more room after coming back from the
+   Christmas holidays.
+
+2. The last Kubernetes release of a year should be finished by the middle of
+   December.
+
+3. A Kubernetes release cycle has a length of of ~15 weeks.
+
+4. Events like KubeCon will be considered as blocked from development or
+   decision-making. SIG Release will also consider the week before and after
+   the event in the same way.
+
+5. An explicit break of at least two weeks between each release cycle will be
+   enforced.
+
+   This does not mean that zero development can happen during that time.
+   Rather, SIG Release will use this time to do the release retrospective and
+   plan for the next cycle.
+
+### Feedback survey
+
+Each minor Kubernetes release will be an experience survey, which will include
+questions around the release cadence.
```

Comment:
we need objective metrics, subjective only is not useful ... unless you have people that is expert in surveys, the audience sample is significative, .... that is a science

Reply:
I'm wondering if we try to over-analyze the situation. Right now we do not actively track those metrics over the past releases. What we do, is using "alerting moments" (let's say we had many issues in the previous test-freeze) to adjust the next release schedule. This already provides us a fair amount of flexibility for such a rapid growing community and project.

What should we do with those metrics? I doubt that we use the metrics for 1.14 to make decisions about the 1.22 release cycle. We would have to analyze so many more factors beside the plain stats, for example the size of the features, the amount of PRs merged in general or which enhancements have been tracked and how many people were working on them.

Reply:
> I doubt that we use the metrics for 1.14 to make decisions about the 1.22 release cycle

I'm not meaning that,  metrics are for avoid this happening again, we can't change the past, but we should learn from it. If we define the 4 or 5 points we want to improve, we can measure it and take decisions based on data. This is a polemic decision because nobody can demonstrate their point, everything is vague and based on feelings and experiences ... 

> We would have to analyze so many more factors beside the plain stats,

Is the other way around, you don't need hundred metrics, you just need to pick a small number of them that are able to demonstrate your hypothesis, and iterate, revisit, .... and next time we can say "hey, we need to reduce/increase the release cadence because X is higher, Y is lower and Z is increasing ...."

Per example, in this case we are claiming that the change in the release cadence will:
- give more time to work on the features and not rush it, so `number of exceptions requests` can give us an idea if this is right or not, if we have more exception requests after the change that before it means clearly that argument was wrong
- improve quality because sigs will have more time for testing, tech debt, ... so `number of issues open during code/test freeze` will tell us if that is true or we keep pushing to the latest .
- .....

Reply:
@aojea looks like you already have a plan in mind how to approach this kind of metrics evaluation. Do we wanna move it into a dedicated issue in k/sig-release and are you happy to work on this topic?

Reply:
> @aojea looks like you already have a plan in mind how to approach this kind of metrics evaluation

That is based on experience, after lots of tries and success, but that worked fine for us in other similar project 

> move it into a dedicated issue in k/sig-release and are you happy to work on this topic?

sure, I need some time, but I will do it

Thanks

Reply:
Thank you! I'll resolve this discussion for now and we follow up.

Reply:
Created a follow-up issue in SIG release: https://github.com/kubernetes/sig-release/issues/1527 Feel free to add your thoughts there, too.